youtube_dl/extractor/yle.py

```diff
@@ -0,0 +1,278 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import base64
+import re
+
+from .common import InfoExtractor
+from ..compat import (
+    compat_HTTPError,
+    compat_urllib_parse,
+    compat_urlparse,
+)
+from ..utils import(
+    bytes_to_intlist,
+    intlist_to_bytes,
+    parse_duration,
+    parse_iso8601,
+    strip_jsonp,
+    ExtractorError,
+)
+from ..aes import (
+    aes_cfb_decrypt,
+    BLOCK_SIZE_BYTES,
+)
+
+
+class YLEAreenaIE(InfoExtractor):
+    _VALID_URL = r'^http://(?P<host>areena|arenan)\.yle\.fi/(?P<id>[\d-]+)'
+    _PROTOCOLS = ['RTMPE', 'HDS']
+    _AES_KEY = 'yjuap4n5ok9wzg43'
+    _PROGRAM_URL = 'http://player.yle.fi/api/v1/programs.jsonp?id={pid}'
+    _PROGRAM_KEY = 'program'
+    _EVENT_KEY = 'publicationEvent'
+
+    _TEST = {
+        'url': 'http://areena.yle.fi/1-2825412',
+        'md5': 'c9a0d9cf91a0596126531d6e91114dff',
+        'info_dict': {
+            'id': '1-2825412',
+            'ext': 'flv',
+            'upload_date': '20150525',
+            'description': 'md5:e71da182216dd2e350f9009872c7d72c',
+            'title': 'Ylen aamu-tv: Näin liputat oikein',
+            'timestamp': 1432529700,
+        },
+    }
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        media_id = mobj.group('id')
+
+        lang = 'fi'
+        if mobj.group('host') == 'arenan':
+            lang = 'sv'
+
+        playlist = self._get_playlist(url, media_id)
```

Comment:
Added test case, which returns list of urls (non-empty `data` array). Also `self._live_title` is now used and `is_live` is set `True` for lives.